crates/compiler/builtins/roc/Dict.roc

```diff
@@ -206,6 +225,22 @@ walk = \@Dict { data }, initialState, transform ->
 ##
 ## As such, it is typically better for performance to use this over [Dict.walk]
 ## if returning `Break` earlier than the last element is expected to be common.
+## Below is a real world example
```

Comment:
nit: I wouldn't call this a real world example.

Also, this maybe isn't a good example. There is no guarantee that the result will be 12. Dictionaries can iterate in any order. Just happen to be in order in this case.

Reply:
I changed up the example, I think its ok now

Reply:
Actually, I just realized that my last comment is actually wrong. You think that given I implemented Dict, I would remember these things. Your first example would work 100% of the time. When adding Dict to the standard library, we made it so that iteration is in insertion order. This only changes on `Dict.remove` where we swap the last element with the removed element.

That said, unless you are quite careful, relying on order definitely is more likely to lead to bugs due to `Dict.remove` changing the order. So I still think this is a better example.

Reply:
Ok sweet thats what I initially thought. ty